flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/RecoveredInputChannelTest.java

```
@@ -220,24 +223,24 @@ private void testConcurrentReadStateAndProcessAndRelease(boolean isRemote) throw
 		}
 	}
 
-	private Callable<Void> readRecoveredStateTask(RecoveredInputChannel inputChannel, ChannelStateReader reader) {
+	private Callable<Void> readRecoveredStateTask(RecoveredInputChannel inputChannel, ChannelStateReader reader, boolean verifyRelease) {
 		return () -> {
 			try {
 				inputChannel.readRecoveredState(reader);
-			} catch (CancelTaskException ex) {
-				// within expectation
+			} catch (Throwable t) {
+				assertTrue("The expected exception should only happen in the case of released channel.", verifyRelease && inputChannel.isReleased());
```

As below, it's better not to hide the original exception.